lib/core/connector_settings.rb

```diff
@@ -86,5 +91,12 @@ def initialize(es_response)
       @elasticsearch_response = es_response.with_indifferent_access
       @globals = ElasticConnectorActions.connectors_meta
     end
+
+    def return_if_present(*args)
```

Comment:
The original implementation had a bug that the `||` would cause us to ignore `false` values set in higher precedence locations. This should now return falsey values as long as they are not `nil`